service/controller/key/cluster.go

```diff
@@ -114,6 +116,38 @@ func IsChinaRegion(awsRegion string) bool {
 	return strings.HasPrefix(awsRegion, "cn-")
 }
 
+func IsNewCluster(cluster infrastructurev1alpha2.AWSCluster) bool {
+	// If  condition  list is empty then this is new cluster.
+	if len(cluster.Status.Cluster.Conditions) == 0 {
+		return true
+	}
+	// If versions list is empty then this is a new cluster.
+	if len(cluster.Status.Cluster.Versions) == 0 {
+		return true
+	}
+	// Check if there is transition state Updated or Updating,
+	// this indicates cluster is not new but updated from other version.
+	for _, condition := range cluster.Status.Cluster.Conditions {
+		if condition.Condition == "Updating" {
+			return false
+		}
+		if condition.Condition == "Updated" {
+			return false
+		}
+	}
```

Comment:
We have [HasUpdatingCondition](https://godoc.org/github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2#CommonClusterStatus.HasUpdatingCondition) and [HasUpdatingCondition](https://godoc.org/github.com/giantswarm/apiextensions/pkg/apis/infrastructure/v1alpha2#CommonClusterStatus.HasUpdatingCondition).

Reply:
that is cool